src/main/java/htsjdk/samtools/sra/SRAAlignmentIterator.java

```diff
@@ -149,16 +153,17 @@ private void nextReference() {
         }
 
         try {
+            alignedIterator = null;
+            hasMoreAlignments = false;
+
             hasMoreReferences = referencesChunksIterator.hasNext();
```

Comment:
why is calling nextReference() when referencesChunksIterator.hasNext()==false not an error? I know that this isn't the code you are changing, but I'm curious to understand...